Dump HistorySet MetaData into the correct output csv files

wangcj05 opened the pull request:
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
enable historyset to handle both scalar and vector meta data, and ad check for NAN from code collections
close #986 
##### What are the significant changes in functionality due to this change request?


----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

PaulTalbot-INL commented on tests/framework/PostProcessors/BasicStatistics/test_time_dep_meta.xml:
```diff
@@ -0,0 +1,158 @@
+<?xml version="1.0" ?>
```
Any chance we could trim this case to a minimal necessary test case? This is a lot to comb through.

alfoa commented:
crash :(

wangcj05 commented:
@alfoa This PR should be ok now. I have added some unit test for the meta data loading, printing and reloading. It seems that we do not handle the loading of vector meta data correctly. In the devel, we will ignore any vector meta data. Now this issue should be also fixed.

wangcj05 commented:
@alfoa I have removed the csv check in test. The reason is that sensitivity calculations of this test are sensitive to different platforms, this is because the condition number of this case is >2000. The results are not reliable. Now the test only tests the mechanics.

alfoa commented:
Okay... Approved by Chair of CCB

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/337377/) on 678ba4f : invalidated by @alfoa

alfoa commented:
Checklist passed...

Merging...
